### PR TITLE
Refactor Shareableviz CLI

### PR DESCRIPTION
**Pull request by ravi-kumar-pilla**
## Description

Split PR for https://github.com/kedro-org/kedro-viz/pull/1708

## Development notes

- Updated CLI command `kedro viz deploy` to accept the following options - [--platform, --endpoint, --bucket-name] for shareableviz and introduced a time limit of 5min
- Refactored CLI launcher `click.echo` statements by creating helper function
- Updated pytests

**NOTE:** Please review it after #1739 

## QA notes

- All pytests should pass

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes

### CODE REVIEW

**Review by merelcht — COMMENTED** [review]
I left some small comments, but generally this looks good 👍 
Should any of the changes introduced in this PR or #1739 be added to the release notes?

**Comment by ravi-kumar-pilla** [comment]
> I left some small comments, but generally this looks good 👍 Should any of the changes introduced in this PR or #1739 be added to the release notes?

Thank you @merelcht . All these PRs are mostly internal changes and the outcome is to extend support for Azure, GCP. I will add this in the release note once these PRs are merged to [feature/shareableviz-extended-support](https://github.com/kedro-org/kedro-viz/tree/feature/shareableviz-extended-support)

File: package/kedro_viz/launchers/utils.py [hunk 1]
```diff
@@ -94,13 +93,14 @@ def _start_browser(host: str, port: int):
...
-        while seconds <= VIZ_DEPLOY_TIME_LIMIT:
-            print(f"...Creating your webpage ({seconds}s)", end="\r", flush=True)
-            sleep(1)
-            seconds += 1
-    except KeyboardInterrupt:  # pragma: no cover
-        print("\nCreating your webpage interrupted. Exiting...")
+    elapsed_time = 0
+    while elapsed_time <= timeout and not process_completed.value:
+        print(
+            f"...Creating your build/deploy Kedro-Viz ({elapsed_time}s)",
```
[the remark this answers is not in the document]
Comment (ravi-kumar-pilla) [reply package/kedro_viz/launchers/utils.py:101]: It is rendered as shown in the initial design [here](https://github.com/kedro-org/kedro-viz/issues/1636#issue-1987909628)
...creating your webpage (23s)

Are you suggesting to change it to ...creating your webpage (23seconds) ?

[the remark this answers is not in the document]
Comment (merelcht) [reply package/kedro_viz/launchers/utils.py:101]: ah no I missed the `s`, that looks good 👍

**Review by merelcht — APPROVED** [review]
LGTM 👍

**Review by SajidAlamQB — APPROVED** [review]
Awesome work @ravi-kumar-pilla, looks great!

**Pull request merged**